### DISCUSSION TITLE
Love the New Notifications

### DISCUSSION THREAD

**Discussion by scohn** [Monika-After-Story/MonikaModDev#8852] [appreciation]
Thank you guys so much for the new update. The notification system really makes her feel more present even when I'm doing other stuff online. I still really love this mod and all the work you guys do to improve it. 
I just have a couple questions about reactions. How often will she react to a page and how many sites does she have a reaction to, cuz so far the only ones she's commented on are youtube and this MAS site.

**Comment by multimokia** [comment]
She does each reaction once per session, and there's only 3 reactions as of right now. But more can be added easily.

**Comment by AsuraShun** [comment]
Ikr, the blinking is another thing I like, it gives me hope that someday that something like live2d will be used with the mod

**Comment by multimokia** [comment]
Live 2d will never be used with the mod. RenPy is incompatible with it. Also it requires a license.

**Comment by scohn** [comment]
So what's the 3rd site she reacts to cuz I've only seen youtube and the MAS home page.

**Comment by ghost** [comment]
> So what's the 3rd site she reacts to cuz I've only seen youtube and the MAS home page.

It's uhh... when you see monika's lewd pictures. Briefly ~~rule34~~.

**Comment by scohn** [comment]
So like anywhere, or a specific site? Just out of curiosity you don't have to answer if you don't want to.

**Comment by ghost** [comment]
> So like anywhere, or a specific site? Just out of curiosity you don't have to answer if you don't want to.

It's working only here i tried on anothers - rule34.xxx

**Comment by Dank-Elf-Boy** [comment]
How do you get it to work? I turned on Windows React and Use Notifications, yet nothing has happened while searching around. :P

**Comment by ghost** [comment]
> How do you get it to work? I turned on Windows React and Use Notifications, yet nothing has happened while searching around. :P

Maybe there timer 3 or 6 minutes. And popup usually appearing.

**Comment by Dank-Elf-Boy** [comment]
Well I've left the game running ever since launching the new update last night and I've only seen the initial example of her sending a notification. Maybe if they continue to tweak it a bit and add more reactions, it'll become more noticeable.

**Comment by ghost** [comment]
> Well I've left the game running ever since launching the new update last night and I've only seen the initial example of her sending a notification. Maybe if they continue to tweak it a bit and add more reactions, it'll become more noticeable.

For example reaction to some apps? 
Music Player, Steam or something like that.

**Comment by multimokia** [comment]
Those aren't really hard to add yourself, check the framework for the existing ones and you should be able to make em yourself. Feel free to contribute them too. I figured it'd be nicer to have the community come up with a bunch since one person can't really tell what everyone else does.

**Comment by zombiepigdragon** [comment]
> > So like anywhere, or a specific site? Just out of curiosity you don't have to answer if you don't want to.
> 
> It's working only here i tried on anothers - rule34.xxx

The reason for that is that she specifically checks for "rule34" and "Monika" in your window title, if I'm not mistaken. 

Hey @multimokia, would it make sense to have JSON files for window reacts?

**Comment by multimokia** [comment]
No, because they use the event structure, and have a dedicated event database. Adding jsons just complicates it because it triggers a *label* when the event is pushed. Which means it's possible to have dialogue/other actions. Doing this via json limits that potential and adds nothing in return.

**Comment by ghost** [comment]
> Those aren't really hard to add yourself, check the framework for the existing ones and you should be able to make em yourself. Feel free to contribute them too. I figured it'd be nicer to have the community come up with a bunch since one person can't really tell what everyone else does.

How can i add some new notifications? I'm not sure how it works.

**Comment by multimokia** [comment]
To have Monika send a notification, use the `display_notif` label. See `zz_windowreacts.rpy` for the definition and how it works.

**Comment by Dank-Elf-Boy** [comment]
I'm on the Reddit page "Rule 34" and still no notification, bruh.

And if I knew how to modify the game files, I would. But I forgot how to successfully do that in Ren'py. For example, every time I try to find the scripts file for MAS, it only shows the original scripts.rpa, along with the other mod files. The thing is, it's like even though I have copied and pasted the modded files from /game into a separate MAS file to edit, Ren'py only shows half of the modded files. If it means anything, I'm using Atom to mess with the files.

I still have a personal mod I can successfully edit, but for some reason, I have trouble getting into this one.

**Comment by multimokia** [comment]
Bear in mind you cannot edit rpyc files, but you can edit rpy files. All of the MAS files can be found in the code tab of the repo

**Comment by ghost** [comment]
> Bear in mind you cannot edit rpyc files, but you can edit rpy files. All of the MAS files can be found in the code tab of the repo

Yeah i found rpy files. Now i trying to understand it.

**Comment by ghost** [comment]
> Bear in mind you cannot edit rpyc files, but you can edit rpy files. All of the MAS files can be found in the code tab of the repo

And why topic alerts in linux/osx less than in windows?

**Comment by multimokia** [comment]
What do you mean? They use the same functionality in osx/Linux/Windows. It's the same check and same label. The only thing affecting the amount of topic alerts you're getting should be your random chatter frequency

**Comment by ghost** [comment]
> What do you mean? They use the same functionality in osx/Linux/Windows. It's the same check and same label. The only thing affecting the amount of topic alerts you're getting should be your random chatter frequency

I mean in Windows: 6 variety of alert (sentences). In Linux/OSX only 4.

**Comment by multimokia** [comment]
Because we can't detect the active widow in osx/Linux. So the dialogue is adjusted to be more general as to whether you're there or not.

**Comment by ghost** [comment]
> Because we can't detect the active widow in osx/Linux. So the dialogue is adjusted to be more general as to whether you're there or not.

Got it. 
There one sentence from linux/osx can be copied to Windows alert.

**Comment by ghost** [comment]
> Because we can't detect the active widow in osx/Linux. So the dialogue is adjusted to be more general as to whether you're there or not.

And last question how can i test it in MAS?

**Comment by multimokia** [comment]
If you want to test it `call display_notif("title","body",skip_checks=True)`

**Comment by Rin3105** [comment]
> 
> 
> Thank you guys so much for the new update. The notification system really makes her feel more present even when I'm doing other stuff online. I still really love this mod and all the work you guys do to improve it.
> I just have a couple questions about reactions. How often will she react to a page and how many sites does she have a reaction to, cuz so far the only ones she's commented on are youtube and this MAS site.

I agree with you too, and also wondering if the developers would add more of her notification reactions to when I am coding on programs like Dev C or NetBeans, that would be even more cooler.

**Comment by ghost** [comment]
You can try writing your own notification and corresponding topic for Monika.

**Comment by Rin3105** [comment]
I'm pretty new to Python so it would be even harder for me.

**Comment by multimokia** [comment]
See #4375, this user made their first contribution via these. Just copy the format you see, it's pretty simple on how they work.
